src/main/java/net/mcreator/workspace/Workspace.java

```diff
@@ -187,13 +210,38 @@ public void addModElement(ModElement element) {
 		if (!mod_elements.contains(element)) { // only add this mod element if it is not already added
 			element.reinit(this); // if it is new element, it now probably has icons so we reinit modicons
 			mod_elements.add(element);
+
+			if (element.getGeneratableElement() instanceof ITabContainedElement tabElement) {
+				TabEntry tabEntry = tabElement.getCreativeTab();
+				if (tabEntry != null && !tabEntry.getUnmappedValue().equals("No creative tab entry"))
+					tab_element_order.get(tabEntry.getUnmappedValue()).add(element.getName());
+			}
+
 			markDirty();
 		} else {
 			LOG.warn(
 					"Trying to add existing mod element: " + element.getName() + " of type " + element.getTypeString());
 		}
 	}
 
+	public void updateModElementTab(ModElement element) {

```

Comment:
Code in methods like this one needs some comments. And not comments that 1:1 explain what code does, but comments that explain what is being checked/done on an upper level that help external reader understand the idea behind the code

Reply:
Would this amount suffice? Other changed lines look too simple to be described other than 1:1.

Reply:
Thanks for adding more comments. I will let you know for specific places as I review the code